### PR TITLE
feat(api-request-builder): add missing services to cover all API endpoints

### PR DESCRIPTION
**Pull request by emmenko**
#### Summary
Define remaining services for API endpoints.

Note: there are a few extra things that should also be supported but they can be done separately.

#### Todo
- Tests
    - [x] Unit

### CODE REVIEW

**Comment by nkuehn** [comment]
Wonderful!   

From a "build a shopfront" use case you probably also want to be able to access the project metadata under  the root url an want to trigger the stuff that's a path below the customer like http://dev.commercetools.com/http-api-projects-customers.html#change-customers-password  and the password reset token.  Not a high priority though.

File: packages/commercetools-api-request-builder/src/create-service.js [hunk 1]
```diff
@@ -92,18 +92,27 @@ export default function createService (
     // Pass some options to further configure the URI:
     // - `projectKey`: will prefix the URI with the given projectKey
     build (options: BuildOptions = {}): string {
+      // FIXME: make it an option of `createRequestBuilder`
```
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/create-service.js]: Should resolve this before merging or remove it & open an issue about it so we can track it there.

File: packages/commercetools-api-request-builder/src/create-service.js [hunk 2]
```diff
@@ -92,18 +92,27 @@ export default function createService (
...
+        getIdOrKey(this.params) +
         (queryParams ? `?${queryParams}` : '')
 
       setDefaultParams.call(this)
       return uri
     },
   })
 }
+
+function getIdOrKey (params: Object): string {
```
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/create-service.js]: This looks odd, why not two functions and put the `if - else` logic where needed?

File: packages/commercetools-api-request-builder/src/default-services.js [hunk 3]
```diff
@@ -57,6 +155,7 @@ export default {
       features.projection,
     ],
   },
+  // TODO: productSuggestions
```
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/default-services.js]: Would also put this is an issue.

File: packages/commercetools-api-request-builder/src/query-id.js [hunk 4]
```diff
@@ -6,11 +6,35 @@
  * @throws If `id` is missing.
  * @return {Object} The instance of the service, can be chained.
  */
-// eslint-disable-next-line import/prefer-default-export
```
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/query-id.js]: ⬆️

File: packages/commercetools-api-request-builder/src/query-projection.js [hunk 5]
```diff
@@ -11,3 +11,9 @@ export function staged (isStaged: boolean = true): Object {
   this.params.staged = isStaged
   return this
 }
+
+// TODO: add price selection query params
```
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/query-projection.js]: Convert to issue.

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 1 above)
Comment (emmenko) [inline packages/commercetools-api-request-builder/src/create-service.js]: We should create an issue yes, but this should be done separately and it's a breaking change

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 2 above)
Comment (emmenko) [inline packages/commercetools-api-request-builder/src/create-service.js]: ??

File: packages/commercetools-api-request-builder/src/default-services.js (hunk 3 above)
Comment (emmenko) [inline packages/commercetools-api-request-builder/src/default-services.js]: I've added it in the project backlog, which can be converted into an issue

https://github.com/commercetools/nodejs/projects/1

File: packages/commercetools-api-request-builder/src/query-projection.js (hunk 5 above)
Comment (emmenko) [inline packages/commercetools-api-request-builder/src/query-projection.js]: Also in the backlog ;)

https://github.com/commercetools/nodejs/projects/1

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 2 above)
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/create-service.js]: Oh nvm, this function confused me 😅 .

**Comment by emmenko** [comment]
@nkuehn 

> From a "build a shopfront" use case you probably also want to be able to access the project metadata under the root url an want to trigger the stuff that's a path below the customer like http://dev.commercetools.com/http-api-projects-customers.html#change-customers-password and the password reset token. Not a high priority though.

Yeah, although this package mostly cares about building the query parameters. If you just have an endpoint like `/customers/password` without any query parameters you might as well just type it yourself ;)
But I still haven't fully decided how to deal with such simple (special) cases

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 2 above)
Comment (emmenko) [inline packages/commercetools-api-request-builder/src/create-service.js]: 😄

File: packages/commercetools-api-request-builder/src/query-id.js (hunk 4 above)
Comment (hisabimbola) [inline packages/commercetools-api-request-builder/src/query-id.js]: Not needed anymore I guess.

File: packages/commercetools-api-request-builder/src/create-service.js [hunk 6]
```diff
@@ -92,18 +92,27 @@ export default function createService (
     // Pass some options to further configure the URI:
     // - `projectKey`: will prefix the URI with the given projectKey
     build (options: BuildOptions = {}): string {
```
Comment (hisabimbola) [inline packages/commercetools-api-request-builder/src/create-service.js]: @Siilwyn doesn't this have something to do with the error you were having this evening?

File: packages/commercetools-api-request-builder/src/create-service.js [hunk 7]
```diff
@@ -92,18 +92,27 @@ export default function createService (
     // Pass some options to further configure the URI:
     // - `projectKey`: will prefix the URI with the given projectKey
```
Comment (hisabimbola) [inline packages/commercetools-api-request-builder/src/create-service.js]: @Siilwyn doesn't this have something to do with the error you were having this evening?

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 1 above)
Comment (hisabimbola) [inline packages/commercetools-api-request-builder/src/create-service.js]: So it's going to be remove 😑 (and probably a better context)

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 2 above)
Comment (hisabimbola) [inline packages/commercetools-api-request-builder/src/create-service.js]: You will get use to **type** function soon @Siilwyn 😜

File: packages/commercetools-api-request-builder/src/create-service.js (hunk 7 above)
Comment (Siilwyn) [inline packages/commercetools-api-request-builder/src/create-service.js]: :o it is! That fixed it, thank you! :3

**Comment by emmenko** [comment]
Alright, TODOs / FIXMEs have been converted to issues.

**Pull request merged**